app/scripts/models/reliers/oauth.js

```diff
@@ -61,6 +66,83 @@ define([
       return true;
     },
 
+    /**
+     * OAuth reliers can opt in to fetch relier-specific keys.
+     */
+    isKeyFetchEnabled: function () {
+      if (this.get('keys')) {
+        return true;
+      }
+      return Relier.prototype.isKeyFetchEnabled.call(this);
+    },
+
+    deriveRelierKeys: function (uid, keys) {
```

Comment:
Nit: does it make sense to move most of this new stuff into its own module / file? and just use `deriveRelierKeys` in the model?